src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,45 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  static removeFromAlive(animal) {
+    Animal.alive = Animal.alive.filter((a) => a !== animal);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
-class Carnivore extends Animal {
-  // write your code here
+class Carnivore extends Herbivore {
+  bite(target) {
+    if (
+      target instanceof Carnivore ||
+      (target.hidden && target instanceof Herbivore)
```

Comment:
The `instanceof` check for `Herbivore` should be outside of the parentheses that check for the `hidden` property to ensure that the `bite` method only affects `Herbivore` instances.